airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -0,0 +1,257 @@
+        )
+        # backoff handlers wrap _send, so it will always return a response
+        response = backoff_handler(user_backoff_handler)(request, request_kwargs)
+
+        return response
+
+    def _send(self, request: requests.PreparedRequest, request_kwargs: Optional[Mapping[str, Any]] = None) -> requests.Response:
+
+        self._logger.debug(
+            "Making outbound API request", extra={"headers": request.headers, "url": request.url, "request_body": request.body}
+        )
+
+        if request_kwargs is None:
+            request_kwargs = {}
+
+        try:
+            response: requests.Response = self._session.send(request, **request_kwargs)
+            self._response_decoder.validate_response(response)
+        except requests.RequestException as exc:
+            response_action = ResponseAction.RETRY
+            failure_type = FailureType.transient_error
+            response_error_message = f"Request to {request.url} failed with exception: {exc}"
+            self._logger.debug(response_error_message)
+            exc = exc
+            response = None
+        else:
+            response_action, failure_type, response_error_message = self._error_handler.interpret_response(response=response)
+            exc = None
+
+        # Evaluation of response.text can be heavy, for example, if streaming a large response
+        # Do it only in debug mode
+        if response is not None:
+            if self._logger.isEnabledFor(logging.DEBUG):
+                self._logger.debug(
+                    "Receiving response", extra={"headers": response.headers, "status": response.status_code, "body": response.text}
+                )
+
+            if response_action == ResponseAction.FAIL:
+                error_message = (
+                    f"Request failed to {request.url} with error {exc}"
+                    if exc
+                    else response_error_message
+                    or f"Request to {response.request.url} failed with status code {response.status_code} and error message {self._error_message_parser.parse_response_error_message(response)}"
+                )
+                # TODO: Provide better internal/external error messaging
+                raise AirbyteTracedException(
+                    internal_message=error_message,
+                    message=error_message,
+                    failure_type=failure_type,
+                )
+
+            if response_action == ResponseAction.IGNORE:
+                self._logger.info(
+                    f"Ignoring response with status code {response.status_code} for request to {request.url}"
+                    if response
+                    else f"Ignoring response for request to {request.url} with error {exc}"
+                )
+
+        # TODO: Consider dynamic retry count depending on subsequent error codes
+        if response_action == ResponseAction.RETRY:
```

Comment:
This condition and the `if not response_action` can be `elif` since they're all mutually exclusive

Reply:
Updated.